src/java.base/share/classes/java/lang/StringCoding.java

```diff
@@ -25,495 +25,21 @@
-    }
-
-    static byte[] encode(Charset cs, byte coder, byte[] val) {
-        if (cs == UTF_8) {
-            return encodeUTF8(coder, val, true);
-        }
-        if (cs == ISO_8859_1) {
-            return encode8859_1(coder, val);
-        }
-        if (cs == US_ASCII) {
-            return encodeASCII(coder, val);
-        }
-        CharsetEncoder ce = cs.newEncoder();
-        // fastpath for ascii compatible
-        if (coder == LATIN1 && (((ce instanceof ArrayEncoder) &&
-                                 ((ArrayEncoder)ce).isASCIICompatible() &&
-                                 !hasNegatives(val, 0, val.length)))) {
-            return Arrays.copyOf(val, val.length);
-        }
-        int len = val.length >> coder;  // assume LATIN1=0/UTF16=1;
-        int en = scale(len, ce.maxBytesPerChar());
-        byte[] ba = new byte[en];
-        if (len == 0) {
-            return ba;
-        }
-        ce.onMalformedInput(CodingErrorAction.REPLACE)
-          .onUnmappableCharacter(CodingErrorAction.REPLACE)
-          .reset();
-        if (ce instanceof ArrayEncoder) {
-            int blen = (coder == LATIN1 ) ? ((ArrayEncoder)ce).encodeFromLatin1(val, 0, len, ba)
-                                          : ((ArrayEncoder)ce).encodeFromUTF16(val, 0, len, ba);
-            if (blen != -1) {
-                return safeTrim(ba, blen, true);
-            }
-        }
-        boolean isTrusted = cs.getClass().getClassLoader0() == null ||
-                            System.getSecurityManager() == null;
-        char[] ca = (coder == LATIN1 ) ? StringLatin1.toChars(val)
-                                       : StringUTF16.toChars(val);
-        ByteBuffer bb = ByteBuffer.wrap(ba);
-        CharBuffer cb = CharBuffer.wrap(ca, 0, len);
-        try {
-            CoderResult cr = ce.encode(cb, bb, true);
-            if (!cr.isUnderflow())
-                cr.throwException();
-            cr = ce.flush(bb);
-            if (!cr.isUnderflow())
-                cr.throwException();
-        } catch (CharacterCodingException x) {
-            throw new Error(x);
-        }
-        return safeTrim(ba, bb.position(), isTrusted);
-    }
-
-    static byte[] encode(byte coder, byte[] val) {
-        Charset cs = Charset.defaultCharset();
-        if (cs == UTF_8) {
-            return encodeUTF8(coder, val, true);
-        }
-        if (cs == ISO_8859_1) {
-            return encode8859_1(coder, val);
-        }
-        if (cs == US_ASCII) {
-            return encodeASCII(coder, val);
-        }
-        StringEncoder se = deref(encoder);
-        if (se == null || !cs.name().equals(se.cs.name())) {
-            se = new StringEncoder(cs, cs.name());
-            set(encoder, se);
-        }
-        return se.encode(coder, val);
-    }
+    static final Charset ISO_8859_1 = sun.nio.cs.ISO_8859_1.INSTANCE;
+    static final Charset US_ASCII = sun.nio.cs.US_ASCII.INSTANCE;
+    static final Charset UTF_8 = sun.nio.cs.UTF_8.INSTANCE;
```

Comment:
These could move to String also, if there is a benefit them being local.
Otherwise, the uses in String could refer directly to the INSTANCEs in the sun.nio.cs... classes.